lib/samlr/tools.rb

```diff
@@ -105,5 +105,43 @@ def self.validate(options = {})
         end
       end
     end
+
+    # Tries to parse the SAML request, returns nil if no data passed.
+    # First, it assumes it to be Base64 encoded.
+    # If this fails, it subsequently attempts to parse the raw input as select IdP's
+    # send that rather than a Base64 encoded value
+    def self.parse(data, request=false)
```

Comment:
I would be more generic and use `compressed: false` instead of `request`. This is also a great time to use an optional variable syntax.